### PR TITLE
modal pop up

### PR DESCRIPTION
**Pull request by reshma-qburst**
This branch has:

1) Modal pop up
2) Few template changes
3) Angular table changes - added bootstrap css

### CODE REVIEW

**Comment by nishintn** [comment]
@reshma-qburst ,
Can you share the gh page link?

**Comment by reshma-qburst** [comment]
It's not up yet.
Will share it in some time.

**Comment by nishintn** [comment]
@reshma-qburst 
Please dont push more changes to this branch. Do only very necessary changes to host this in github. Lets me know once its done.

**Comment by reshma-qburst** [comment]
Ok. Working on the .gitignore file. Will update once done.

**Comment by reshma-qburst** [comment]
gh-pages link : 
https://reshma-qburst.github.io/key-management-system

Had to include two plugins from node_modules folder.
Was facing issues with gitignore.
Issues resolved.

File: assets/js/build/key-app.min.js [hunk 1]
```diff
@@ -32,9 +32,90 @@ angular.module('keyManagement', ['ui.router','ngTable'])
     	loadJson.getTableDefaultList().then(function(tableData) {
     		$scope.data =  tableData;
 		});
+
+		$scope.showModal = false;
```
Comment (nishintn) [inline assets/js/build/key-app.min.js]: Why do you have business logic in app file? Move this out from here.

File: assets/js/build/key-app.min.js [hunk 2]
```diff
@@ -32,9 +32,90 @@ angular.module('keyManagement', ['ui.router','ngTable'])
...
+	    $scope.toggleModal = function(){
+	        $scope.showModal = !$scope.showModal;
+	    };
+
+	    $scope.isValidationModeDisabled = true;
+	    $scope.isSizeOfKeyDisabled = true;
+
+	    $scope.change = function(val) {
+	        if(val === "t1" || val === "t3"){
```
Comment (nishintn) [inline assets/js/build/key-app.min.js]: What is t1, t2...?

File: assets/js/build/key-app.min.js [hunk 3]
```diff
@@ -32,9 +32,90 @@ angular.module('keyManagement', ['ui.router','ngTable'])
...
+	        	$scope.isValidationModeDisabled = true;
+	        }
+	    };
+
+	    $scope.changeValidationMode = function(){
+	    	$scope.isSizeOfKeyDisabled = false;
+	    };
+
+	    $scope.addDataToTable = function(){
+	    	console.log($scope.descriptionNew);
```
Comment (nishintn) [inline assets/js/build/key-app.min.js]: Console and alert sts are not allowed to push to githubs.

File: assets/js/build/key-app.min.js [hunk 4]
```diff
@@ -32,9 +32,90 @@ angular.module('keyManagement', ['ui.router','ngTable'])
...
+	    	alert("Generated Key  ::: "+Math.random().toString(36).substr(2, 16));
+	    };
+
+	    $scope.beforeRender = function ($dates) {
```
Comment (nishintn) [inline assets/js/build/key-app.min.js]: This ia util function which can be re used. Move it to a factory/ service.

File: assets/js/build/key-app.min.js [hunk 5]
```diff
@@ -32,9 +32,90 @@ angular.module('keyManagement', ['ui.router','ngTable'])
...
+		        }
+		    }  
+		}
     }]);
 })();
 
+(function(){
+"use strict";
+
+  angular.module('keyManagement').directive('modal', function () {
```
Comment (nishintn) [inline assets/js/build/key-app.min.js]: Modal is a generic name. What if you have another modal? Please use meaningful names.

File: assets/js/build/key-app.min.js (hunk 2 above)
Comment (reshma-qburst) [inline assets/js/build/key-app.min.js]: t1, t2 represents type of key. Have already made the change in new branch (addTableData).

File: assets/js/build/key-app.min.js (hunk 5 above)
Comment (reshma-qburst) [inline assets/js/build/key-app.min.js]: Shall change it.

Comment (reshma-qburst) [inline assets/js/build/key-app.min.js]: Added a customised modal name.

**Comment by reshma-qburst** [comment]
Have worked on the added comments & made this branch stable.

**Pull request merged**